src/v/transform/logging/tests/log_manager_test.cc

```diff
@@ -0,0 +1,273 @@
+
+namespace {
+
+using namespace std::chrono_literals;
+using manager_t = transform::logging::manager<ss::manual_clock>;
+
+std::chrono::milliseconds flush_interval() {
+    return config::shard_local_cfg()
+      .data_transforms_logging_flush_interval_ms();
+}
+
+void advance_clock(ss::manual_clock::duration dur = flush_interval()) {
+    ss::manual_clock::advance(dur);
+    tests::drain_task_queue().get();
+}
+
+class fake_client : public transform::logging::client {
+public:
+    fake_client() = default;
+
+    ss::future<>
+    write(model::transform_name_view, ss::chunked_fifo<iobuf> events) override {
+        std::move(events.begin(), events.end(), std::back_inserter(_logs));
+        return ss::now();
+    }
+
+    const ss::chunked_fifo<iobuf>& logs() const { return _logs; }
+
+private:
+    ss::chunked_fifo<iobuf> _logs;
+};
+
+} // namespace
+
+class TransformLogManagerTest : public ::testing::Test {
+public:
+    void SetUp() override {
+        auto sink = std::make_unique<fake_client>();
+        _client = sink.get();
+        _manager = std::make_unique<manager_t>(
+          model::node_id{0}, std::move(sink));
+        start_manager();
+    }
+
+    void SetUp(size_t buffer_capacity, size_t line_max) {
+        config::shard_local_cfg()
+          .data_transforms_logging_buffer_capacity_bytes.set_value(
+            buffer_capacity);
+        config::shard_local_cfg()
+          .data_transforms_logging_line_max_bytes.set_value(line_max);
+        SetUp();
+    }
+
+    void TearDown() override {
+        _client = nullptr;
+        _manager->stop().get();
+        _manager.reset();
+    }
+
+    void start_manager() { _manager->start().get(); }
+    void stop_manager() { _manager->stop().get(); }
+
+    void enqueue_log(
+      ss::log_level lvl,
+      model::transform_name_view name,
+      std::string_view msg) {
+        _manager->enqueue_log(lvl, name, msg);
+    }
+
+    void enqueue_log(std::string_view msg) {
+        enqueue_log(
+          ss::log_level::info, model::transform_name_view{"foo"}, msg);
+    }
+
+    const ss::chunked_fifo<iobuf>& logs() const { return _client->logs(); }
+
+private:
+    fake_client* _client;
+    std::unique_ptr<manager_t> _manager;
+};
+
+TEST_F(TransformLogManagerTest, EnqueueLogs) {
+    enqueue_log("Hello from some test code!");
+
+    advance_clock();
+    EXPECT_EQ(logs().size(), 1);
+
+    enqueue_log("Hello again from some test code!");
+
+    advance_clock();
+    EXPECT_EQ(logs().size(), 2);
+
+    auto msg = testing::get_message_body(logs().back().copy());
+    EXPECT_TRUE(msg.find("again") != ss::sstring::npos);
+}
+
+TEST_F(TransformLogManagerTest, DISABLED_LastGasp) {
+    constexpr size_t n = 10;
+
+    for (int i = 0; i < n; ++i) {
+        enqueue_log("Hello, World!");
+    }
+
+    EXPECT_EQ(logs().size(), 0);
+
+    stop_manager();
+
+    EXPECT_EQ(logs().size(), n);
+    EXPECT_EQ(testing::get_message_body(logs().back().copy()), "Hello, World!");
```

Comment:
get_message_body is used alot, can we wrap that in a helper? Does it make sense that this reads something like:

```
EXPECT_EQ(last_log_msg(), "...");
```